### ISSUE TITLE
Implement OnyxSelect Basis

### ISSUE THREAD

**Issue by BoppLi** [SchwarzIT/onyx#762]
### Why?

Provide the basis component for #565 
With this ticket, everything of the select is set up except for the open/close of the flyout and all of the flyout contents.

### Design

[Figma](https://www.figma.com/file/YfEUBOHk4J4nYrk04geswG/onyx---Design-System-Figma-Library?type=design&node-id=1569-34084&mode=design&t=lgLzv7i9PKKCmZad-4)

![Image](https://github.com/SchwarzIT/onyx/assets/151019977/18290478-562f-4ea8-9e8b-21bd867ee282)

### Acceptance criteria

- [x] Figma design is implemented 
  - [x] default, disabled
  - [x] readonly (focus)
  - [x] single/multi select value display, including `summary` vs `preview` mode
  - [x] for `summary` mode: show the value name instead of "1 selected" if only 1 item is selected.
  - [x] for `summary` mode: provide translations for "selected"
  - [x] for `preview` mode: always show the badge even if the list of names is not truncated
- [x] label can be shown or hidden (then `aria-label` is used instead)
- [x] All densities are implemented
- [x] Can be set to skeleton mode
- [x] Can be set to loading: while loading the component behaves just like disabled. Same for the cursor-behavior

**Out-of-scope**

- valid and invalid
- flyout and all related contents

### Definition of Done

- [x] example usage added to alpha-test-app
- [x] covered by component tests
- [x] screenshot tests are updated
- [x] follow-up tickets were created if necessary
  - #763
  - #770
- [x] updated version + documentation is deployed
- [x] Storybook can show the feature
- [x] Set task to `In Approval`

**Approval**

- [Storybook](https://storybook.onyx.schwarz/?path=/docs/components-onyxselect--docs)
- [Docs](https://onyx.schwarz/development/components/OnyxSelect.html)
- [x] approved by designer @jannick-ux
- [x] approved by dev @larsrickert 

### Implementation details

For now, focus on all that is needed to showcase just the "input" part of the select and all of its behaviors.
ModelValue Type will be adjusted for comboBox/listBox later.

#### Applicable ARIA Pattern

[Combobox](https://w3c.github.io/aria/#combobox)
[Listbox](https://w3c.github.io/aria/#listbox)

**Comment by BoppLi** [comment]
![Image](https://github.com/SchwarzIT/onyx/assets/151019977/061991dd-4835-407f-91fd-848c2f711efd)

**Comment by larsrickert** [comment]
@BoppLi Minor points:

1. Badge is merged so you can remove your custom badge implementation
2. Property `multiple`: If an object is passed, `textMode` should be required, otherwise you could pass an empty object. Or we change the type to `boolean | MultiselectTextMode` or `false | MultiselectTextMode` so you have to set a multiselect mode and its more explicit 
3. There is an open TODO for the skeleton density, do you want to resolve it or do we have a follow up ticket?
4. The [CaniUse score](https://caniuse.com/?search=lh%20unit) of the `lh` unit is only 88%. Since we have a `$line-height` CSS variable already, we should use this instead imho

**Comment by BoppLi** [comment]
1) done in Majas PR https://github.com/SchwarzIT/onyx/pull/831
2) I discussed this with @JoCa96 and he voted for allowing passing an empty object. the reason is that the config can easily grow in the future. And if it is an empty object, we use the documented default behavior of textMode.
3) the TODO is already done, just forgot to remove it ^^
4) thx! took it over from the Input, there it already got fixed in the meantime it seems ^^

3) + 4) done in [PR 835](https://github.com/SchwarzIT/onyx/pull/835)
